File: app/src/main/java/cz/kutner/comicsdb/connector/service/AuthorService.java
package cz.kutner.comicsdb.connector.service;

import java.util.List;

import cz.kutner.comicsdb.model.Author;
import retrofit.http.GET;
import retrofit.http.Query;
import rx.Observable;

public interface AuthorService {
    @GET("/autorlist.php")
    Observable<List<Author>> listAuthors(@Query("str") int page);

    @GET("/search.php")
    Observable<List<Author>> authorSearch(@Query("searchfor") String keyword);
}

File: app/src/main/java/cz/kutner/comicsdb/connector/service/AuthorService.kt
package cz.kutner.comicsdb.connector.service

import cz.kutner.comicsdb.model.Author
import retrofit.http.GET
import retrofit.http.Query

interface AuthorService {
    @GET("/autorlist.php")
    fun listAuthors(@Query("str") page: Int): List<Author>

    @GET("/search.php")
    fun authorSearch(@Query("searchfor") keyword: String): List<Author>
}
